extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -0,0 +1,226 @@
+      tempShadowContainer.appendChild(shadowStyle);
+      tempShadowContainer.appendChild(shadowContent);
+      // set the shadowRoot_ equal to that container
+      this.shadowRoot_ = tempShadowContainer;
+    }
+  }
+
+  /**
+   * Finds the prompt and options content
+   * and adds it to the shadow DOM.
+   * @private
+   */
+  attachContent_() {
+    // TODO(jackbsteinberg): Optional prompt behavior must be implemented here
+    const prompt = this.element.children[0];
+    // First child must be heading h1-h3
+    if (!['h1', 'h2', 'h3'].includes(prompt.tagName.toLowerCase())) {
+      user().error(
+        TAG,
+        'The first child must be a heading element <h1>, <h2>, or <h3>'
+      );
+    }
+
+    prompt.classList.add('i-amp-story-quiz-prompt');
+
+    const options = toArray(this.element.querySelectorAll('option'));
+    if (options.length < 2 || options.length > 4) {
+      user().error(TAG, 'Improper number of options');
+    }
+
+    this.shadowRoot_
+      .querySelector('.i-amp-story-quiz-prompt-container')
+      .appendChild(prompt);
+
+    options.forEach((option, index) => this.configureOption_(option, index));
+
+    if (this.element.children.length !== 0) {
+      user().error(TAG, 'Too many children');
+    }
+
+    if (this.shadowRoot_ !== this.element.shadowRoot) {
+      this.element.append(this.shadowRoot_);
+    }
+  }
+
+  /**
+   * Creates an option container with option content,
+   * adds styling and answer choices,
+   * and adds it to the shadow DOM.
+   * @param {HTMLOptionElement} option
+   * @param {number} index
+   * @private
+   */
+  configureOption_(option, index) {
+    const html = htmlFor(option);
+    const convertedOption = html`
+      <span class="i-amp-story-quiz-option">
+        <span class="i-amp-story-quiz-answer-choice"></span>
+      </span>
+    `;
+
+    // Fill in the answer choice
+    convertedOption.querySelector(
+      '.i-amp-story-quiz-answer-choice'
+    ).textContent = answerChoiceOptions[index];
+
+    // Transfer the option information into a span then remove the option
+    const optionText = document.createTextNode(option.textContent);
+    convertedOption.append(optionText);
+
+    if (option.hasAttribute('correct')) {
+      convertedOption.setAttribute('correct', 'correct');
+    }
+    option.remove();
+
+    // Add the option to the shadow root
+    this.shadowRoot_
+      .querySelector('.i-amp-story-quiz-option-container')
+      .appendChild(convertedOption);
+  }
+
+  /**
+   * Attaches functions to each option to handle state transition.
+   * @private
+   */
+  attachOptionActionHandlers_() {
+    const options = toArray(
+      this.shadowRoot_.querySelectorAll('.i-amp-story-quiz-option')
+    );
+
+    // Add a click listener to the element to handle option selection and navigation via tapping the prompt
+    this.element.addEventListener('click', e => {
+      let resolved = false;
+      e.composedPath().forEach(p => {
```

Comment:
I'm not 100% sure of the browser support for this, as Shadow DOM has two different APIs (v0 and v1). Also, it won't work when there's no shadow DOM support.

If you listen for events on `this.quizEl_` (cf previous comment), you should be able to remove the `composedPath()` here.

Reply:
Ok thanks for the suggestion, refactored to use this method.

Additionally, I removed the prompt check for the moment, I think I'd like to wait and pull that change out into a PR where I change the interaction permission check like Jon suggested.